chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/developer/attestation/CheckInReportValidation.java

```diff
@@ -0,0 +1,184 @@
+
+import gov.healthit.chpl.domain.CertificationCriterion;
+import gov.healthit.chpl.form.Form;
+import gov.healthit.chpl.search.domain.ListingSearchResult;
+import gov.healthit.chpl.service.CertificationCriterionService;
+import gov.healthit.chpl.service.realworldtesting.RealWorldTestingCriteriaService;
+import gov.healthit.chpl.util.ErrorMessageUtil;
+
+@Component
+public class CheckInReportValidation {
+    private static final String RWT_VALIDATION_TRUE = "Has listing(s) with RWT criteria";
+    private static final String RWT_VALIDATION_FALSE = "No listings with RWT criteria";
+    private static final String ASSURANCES_VALIDATION_TRUE = "Has listing(s) with Assurances criteria (b)(6) or (b)(10)";
+    private static final String ASSURANCES_VALIDATION_FALSE = "No listings with Assurances criteria (b)(6) or (b)(10)";
+    private static final String API_VALIDATION_TRUE = "Has listing(s) with API criteria (g)(7)-(g)(10)";
+    private static final String API_VALIDATION_FALSE = "No listings with API criteria (g)(7)-(g)(10)";
+
+    private ErrorMessageUtil errorMessageUtil;
+
+    private List<CertificationCriterion> assurancesCriteria;
+    private List<CertificationCriterion> apiCriteria;
+    private List<CertificationCriterion> rwtCriteria;
+
+    @Autowired
+    public CheckInReportValidation(RealWorldTestingCriteriaService realWorldTestingCriteriaService,
+            CertificationCriterionService certificationCriterionService,
+            ErrorMessageUtil errorMessageUtil,
+            @Value("${assurancesCriteriaKeys}") String[] assurancesCriteriaKeys,
+            @Value("${apiCriteriaKeys}") String[] apiCriteriaKeys) {
+
+        this.errorMessageUtil = errorMessageUtil;
+
+        Integer currentYear = Calendar.getInstance().get(Calendar.YEAR);
+        rwtCriteria = realWorldTestingCriteriaService.getEligibleCriteria(currentYear);
+        assurancesCriteria = Arrays.asList(assurancesCriteriaKeys).stream()
+                .map(key -> certificationCriterionService.get(key))
+                .collect(Collectors.toList());
+        apiCriteria = Arrays.asList(apiCriteriaKeys).stream()
+                .map(key -> certificationCriterionService.get(key))
+                .collect(Collectors.toList());
+    }
+
+    public String getRealWorldTestingValidationMessage(List<ListingSearchResult> allActiveListingsForDeveloper) {
+        if (isRealWorldTestingValid(allActiveListingsForDeveloper)) {
+            return errorMessageUtil.getMessage("attestatation.checkInReport.rwtValidationTrue");
+        } else {
+            return errorMessageUtil.getMessage("attestatation.checkInReport.rwtValidationFalse");
+        }
+    }
+
+    public String getAssurancesValidationMessage(List<ListingSearchResult> allActiveListingsForDeveloper) {
+        if (isAssurancesValid(allActiveListingsForDeveloper)) {
+            return errorMessageUtil.getMessage("attestatation.checkInReport.assurancesValidationTrue");
```

Comment:
Thanks for changing these but now I think a change is needed in errors.properties to change the same property name